visualization/app/codeCharta/util/treeMapGenerator.ts

```diff
@@ -92,6 +129,10 @@ function getSquarifiedTreeMap(map: CodeMapNode, state: State): SquarifiedTreeMap
 		mapHeight = state.treeMap.mapSize * 2
 	}
 
+	// nodesPerSide is just an estimation.
+	// We do not know the exact amount,
+	// because the treemap algorithm is/must be executed with an initial width and height afterwards.
+	// TODO If it is wrong some buildings might be cut off.
 	const width = mapWidth + nodesPerSide * state.dynamicSettings.margin
 	const height = mapHeight + nodesPerSide * state.dynamicSettings.margin
```

Comment:
In my opinion the wrong estimation is the result of incorrect rounding after calculating the square root, the standard is that whne converting an integerto float we just cut and not round, e.g. 1.5 = > 1, however Math.ceil would round to the next biggest integer. While we might scale nodes a bit smaller then needed, e.g. ;Math.ceil(1.001) = 2, we would not run the risk of loosing one.

What do you think?

Reply:
That could be a possible solution for the issue #821. Another cause could be the used Padding Scaling Factor which is set to 0.4. If it is increased it can happen that some buildings will disappear.